### PR TITLE
python 3 version

### PR DESCRIPTION
**Pull request by idanre1**
Following suggestions from https://github.com/nfultz/nb2mail/issues/10

1. Fix mime type
2. Fix __init__ issue
3. Pose python3 only constraint: than way when pip is on python2 it will not take latest version. and new versions will automatically be supported.
4. Add example for 3rd party email provider (usually when working on cloud and smtp port is blocked)

### CODE REVIEW

**Comment by idanre1** [comment]
@nfultz  and @0x26res
please review the PR

hopefully now support for python2 and python3 will work seamlessly across different OS setups.

File: README.md [hunk 1]
```diff
@@ -43,6 +44,10 @@ and then run:
 
     jupyter nbconvert --config config.py demo.ipynb
 
+### 3rd party email distributor Example
+Insread of using SMTP to send emails, one can use 3rd party provider.
```
Comment (0x26res) [inline README.md]: Instead.

File: setup.py [hunk 2]
```diff
@@ -8,6 +8,7 @@
       author='Neal Fultz',
       author_email='nfultz@gmail.com',
       license='BSD',
+      python_requires='>=3.8, <4',
```
Comment (0x26res) [inline setup.py]: Wouldn't that restrict the ability of someone to install the latest/upcoming version of nb2mail with python 2? I thought you wanted to still support python 2.

**Review by 0x26res — APPROVED** [review]

File: setup.py (hunk 2 above)
Comment (nfultz) [reply setup.py]: would it be easier to rename to `nb2mail3` and leave the current package for legacy users?

File: nb2mail/__init__.py [hunk 3]
```diff
@@ -72,9 +73,10 @@ def _file_extension_default(self):
 
     @default('template_file')
     def _template_file_default(self):
-        return 'mail.tpl'
+        # return os.path.join(os.path.dirname(__file__), "templates", "mail.tpl")
+        "mail.tpl"
```
Comment (nfultz) [inline nb2mail/__init__.py]: no return?

File: setup.py (hunk 2 above)
Comment (idanre1) [reply setup.py]: I am no a package expert.

I thought that if we will have two versions:
* current is 0.6
* new one would be 1.0 (or something else, you decide)

That way on python2 env, pip will automatically will choose 0.6 and on python3 env pip will pick 1.0 version

What do you guys think?

File: README.md (hunk 1 above)
Comment (idanre1) [reply README.md]: Typo is fixed with a new commit.
Thanks!

File: nb2mail/__init__.py (hunk 3 above)
Comment (idanre1) [reply nb2mail/__init__.py]: weird :)
I don't understand how it ran sucessfully without that return.
Anyway I have fixed it and tested to see mails are sent sucessfully with the added `return`
Please review the file after the new commit

File: setup.py (hunk 2 above)
Comment (0x26res) [reply setup.py]: I'm not an expert either. 

We could:
- fork this repo, call it nb2mail3 and publish it as nb2mail3 in pypi
- or consider that we won't support python 2 going forward and leave this as is. If someone really need to use python 2 they can use 0.6.

**Comment by idanre1** [comment]
I have a solution to multi python version support:
https://stackoverflow.com/questions/27376907/how-to-detect-python-version-2-or-3-in-script

I have already tested the new code in python3 and it works.
Does someone have py2env to see if its also working? changes are already pushed.

**Comment by nfultz** [comment]
Thank you for your contribution!

**Comment by idanre1** [comment]
Hi @nfultz 
Will you be willing to create a new release for streamline the PR to pip?

I want to create a new venv and see pip takes the correct code automatically.

**Comment by nfultz** [comment]
I've uploaded to PyPI as v 0.8 - https://pypi.org/project/nb2mail/0.8/

Please do post back if it works for you or not.

**Comment by idanre1** [comment]
Works like a charm!

**Pull request merged**